### DISCUSSION TITLE
Firestore instance has already been started and its settings can no longer be changed.

### DISCUSSION THREAD

**Discussion by bobby-white** [invertase/react-native-firebase#5071] [General]
I am trying to use Firestore natively with Obj-C, and with React Native.

I have a background process (which can be running while app is in foreground too) that I had to write natively, which periodically writes data to a collection in Firestore.
I also read data from Firestore in Javascript.

Both of these features work independently, but the read only happens in certain scenarios.

When I try to use firestore natively and in Javascript at the same time, I get the issue stating that:
`Firestore instance has already been started and its settings can no longer be changed.`

Is there any workarounds to this?

In Obj-c I am accessing Firestore using the Firebase SDKs directly, as follows:
```objective-c
// AppDelegate.m
#import <Firebase.h>
...
if ([FIRApp defaultApp] == nil) {
  [FIRApp configure];
}

// BackgroundTask.m
#import <Firebase.h>
...
[[[[FIRFirestore firestore] collectionWithPath:collectionPath] documentWithPath:documentPath] setData:data];
```

**Comment by bobby-white** [comment]
Is this an appropriate solution? Seems to work...

```objective-c
#import <RNFBFirestore/RNFBFirestoreCommon.h>
...

self.firestore = [RNFBFirestoreCommon getFirestoreForApp:[FIRApp defaultApp]];
...
[[[self.firestore collectionWithPath:collectionPath] documentWithPath:documentPath] setData:data];
```

Am I likely to run into any gotchas later down the road using this solution?

**Reply by mikehardy** [reply]
That's the correct solution I think, it's similar to where I was trying to go with my counsel in #4889 - basically, `@react-native-firebase/firestore` expects to be in charge of the firestore native configuration because we take config variables from `firestore.json` and may configure things based on that or allow overrides etc, so (re-)using the react-native-firebase init code in that way (by calling `getFirestoreForApp` allows all that to happen and then `@react-native-firebase/firestore` is satisfied when you go to use it at the JS level

The only thing I could think of as a trap is that you may not have the opportunity to change firestore pre-usage config settings unless you're careful with order of operations and where you do the configuration. Not a show-stopper by any means, just something to think about
